### PR TITLE
Add parallel integTest for OpenSearch Plugins

### LINKED ISSUE #1951: Automate integ test using RPM bundle.
Currently our integ tests are working only for using tarball distribution. We now have workflow to generate RPM bundle. We need incorporate the automated integ tests for RPM bundle.

### LINKED ISSUE #2025: [Enhancement] Run IntegTest framework with dynamic parallel stages in Jenkins
As of now each integTest takes 1-2 hours to complete on Jenkins, for OS and OSD with similar time range.

This slow down the rapid build we are trying to achieve, and makes the run very heavyweight.

In Jenkins declarative pipeline we can do multi-stage dynamic trigger like [this](https://stackoverflow.com/questions/53234490/dynamic-parallel-stages-in-jenkins-pipeline-outside-script-block):

```
pipeline{
  agent any
  stages{
    stage('CI'){
      steps{
        script {
            doDynamicParallelSteps()
        }
      }
    }
  }
}

def doDynamicParallelSteps(){
  tests = [:]
  for (f in findFiles(glob: '**/html/*.html')) {
    tests["${f}"] = {
      node {
        stage("${f}") {
          echo '${f}'
        }
      }
    }
  }
  parallel tests
}

```

This would help us speed up the runs and makes the test lightweight.
We probably also need to check whether `--component` is supported in integTest framework so we can only run 1 test per stage and each stage have a separate docker container.

Thanks.

### PR DESCRIPTION
**Pull request by peterzhuamazon**
Signed-off-by: Peter Zhu <zhujiaxi@amazon.com>

### Description
Add parallel integTest for OpenSearch Plugins

### Issues Resolved
#2025

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: jenkins/opensearch/integ-test.jenkinsfile [hunk 1]
```diff
@@ -58,56 +54,88 @@ pipeline {
...
-                        )
-                    }
+                    //script {
+                    //    uploadTestResults(
+                    //        buildManifestFileName: BUILD_MANIFEST,
+                    //        jobName: JOB_NAME,
+                    //        buildNumber: BUILD_ID
+                    //    )
+                    //}
                     postCleanup()
```
Comment (zelinh) [inline jenkins/opensearch/integ-test.jenkinsfile]: I think you forget to uncomment these and the message section below.

Comment (peterzhuamazon) [reply jenkins/opensearch/integ-test.jenkinsfile]: Fixed.

File: jenkins/opensearch-dashboards/integ-test.jenkinsfile [hunk 2]
```diff
@@ -57,8 +53,8 @@ pipeline {
         stage('detect docker image + args') {
             agent {
                 docker {
-                    label 'Jenkins-Agent-al2-x64-c54xlarge-Docker-Host'
-                    image 'opensearchstaging/ci-runner:centos7-x64-arm64-jdkmulti-node10.24.1-cypress6.9.1-20211028'
+                    label AGENT_LABEL
+                    image 'alpine:latest'
```
Comment (prudhvigodithi) [inline jenkins/opensearch-dashboards/integ-test.jenkinsfile:57]: Hey can we have the alpine tag with fixed version?, as latest every-time can have chances to break, we can go with [3.15.4](https://hub.docker.com/layers/alpine/library/alpine/3.15.4/images/sha256-a777c9c66ba177ccfea23f2a216ff6721e78a662cd17019488c417135299cd89?context=explore)

Comment (prudhvigodithi) [reply jenkins/opensearch-dashboards/integ-test.jenkinsfile:57]: Other than this PR looks good for me.

Comment (peterzhuamazon) [reply jenkins/opensearch-dashboards/integ-test.jenkinsfile:57]: What about `alpine:3`

Comment (peterzhuamazon) [reply jenkins/opensearch-dashboards/integ-test.jenkinsfile:57]: Fixed now.

**Review by prudhvigodithi — APPROVED** [review]

**Pull request merged**